src/heartbeat.jl

```diff
@@ -7,10 +7,13 @@
 import Libdl
 
 const threadid = zeros(Int, 128) # sizeof(uv_thread_t) <= 8 on Linux, OSX, Win
-const zmq_proxy = Ref(C_NULL)
+const zmq_proxy_context = Ref{Context}()
 
 # entry point for new thread
-function heartbeat_thread(sock::Ptr{Cvoid})
+function heartbeat_thread(heartbeat_addr::Cstring)
+    zmq_proxy_context[] = Context()
+    heartbeat = Socket(zmq_proxy_context[], ROUTER)
+    GC.@preserve heartbeat_addr bind(heartbeat, unsafe_string(heartbeat_addr))
```

Comment:
I'd rather create the context and the socket in `start_heartbeat`, and simply pass the `heartbeat` socket pointer to `heartbeat_thread`.

Reply:
The ZMQ docs recommend creating the sockets inside the thread they are used from, and that was doable since the heartbeat socket isn't used anywhere else. But, I don't think there is any advantage to that vs. creating in `init.jl` like it was originally.